flink-connectors/flink-orc/src/main/java/org/apache/flink/orc/OrcFileWriter.java

```diff
@@ -0,0 +1,269 @@
+		return Objects.equals(schema, writer.schema)
+			&& Objects.equals(metaSchema, writer.metaSchema)
+			&& Objects.equals(rowBatch, writer.rowBatch)
+			&& Objects.equals(orcBatchWriter, writer.orcBatchWriter);
+	}
+
+	/**
+	 * The writer that fill the records into the batch.
+	 */
+	private class OrcBatchWriter {
+
+		private List<TypeInformation> typeInfos;
+
+		public OrcBatchWriter(List<TypeInformation> typeInfos) {
+			this.typeInfos = typeInfos;
+		}
+
+		/**
+		 * Fill the record into {@code VectorizedRowBatch}, return false if batch is full.
+		 *
+		 * @param batch  An reusable orc VectorizedRowBatch.
+		 * @param record Input record.
+		 * @return Return false if batch is full.
+		 */
+		private boolean fill(VectorizedRowBatch batch, T record) {
+			// If the batch is full, write it out and start over.
+			if (batch.size == batch.getMaxSize()) {
+				return false;
+			} else {
+				IntStream.range(0, typeInfos.size()).forEach(
+					index -> {
+						setColumnVectorValueByType(typeInfos.get(index), batch, index, batch.size, record);
+					}
+				);
+				batch.size += 1;
+				return true;
+			}
+		}
+
+		private void setColumnVectorValueByType(TypeInformation typeInfo,
+												VectorizedRowBatch batch,
+												int index,
+												int nextPosition,
+												T record) {
+			switch (typeInfo.toString()) {
+				case "Long":
+					LongColumnVector longColumnVector = (LongColumnVector) batch.cols[index];
+					longColumnVector.vector[nextPosition] = (Long) record.getField(index);
+					break;
+				case "Boolean":
+					LongColumnVector booleanColumnVector = (LongColumnVector) batch.cols[index];
+					Boolean bool = (Boolean) record.getField(index);
+					int boolValue;
+					if (bool) {
+						boolValue = 1;
+					} else {
+						boolValue = 0;
+					}
+					booleanColumnVector.vector[nextPosition] = boolValue;
+					break;
+				case "Short":
+					LongColumnVector shortColumnVector = (LongColumnVector) batch.cols[index];
+					shortColumnVector.vector[nextPosition] = (Short) record.getField(index);
+					break;
+				case "Integer":
+					LongColumnVector intColumnVector = (LongColumnVector) batch.cols[index];
+					intColumnVector.vector[nextPosition] = (Integer) record.getField(index);
+					break;
+				case "Float":
+					DoubleColumnVector floatColumnVector = (DoubleColumnVector) batch.cols[index];
+					floatColumnVector.vector[nextPosition] = (Float) record.getField(index);
+					break;
+				case "Double":
+					DoubleColumnVector doubleColumnVector = (DoubleColumnVector) batch.cols[index];
+					doubleColumnVector.vector[nextPosition] = (Double) record.getField(index);
+					break;
+				case "String":
+					BytesColumnVector stringColumnVector = (BytesColumnVector) batch.cols[index];
+					stringColumnVector.setVal(nextPosition, ((String) record.getField(index)).getBytes(StandardCharsets.UTF_8));
+					break;
+				default:
+					throw new IllegalArgumentException("Unsupported column type " + typeInfo);
```

Comment:
You might want to do this type check in constructor as well.

Reply:
Yea. We should check it in constructor as well in order to allow users to find problems early. Will update later.